samples/Gallery/Pages/TreeViewPage.fs

```diff
@@ -2,43 +2,99 @@ namespace Gallery
 
 open System.Diagnostics
 open Avalonia.Controls
+open Avalonia.Interactivity
 open Avalonia.Layout
 open Avalonia.Media
 open Fabulous.Avalonia
 open Fabulous
 
 open type Fabulous.Avalonia.View
 
-module TreeViewPage =
-    type Node = { Name: string; Children: Node list }
+type Node(name, children) =
+    member this.Name = name
+    member this.Children = children
+
+module NodeView =
```

Comment:
@h0lg I have not used `INotifyPropertyChanged` with Fabulous in the past, I normally create a simple MVU component. See my last commit and improve the way you need :)

Reply:
@edgarfgp You may be right. I have seen things like the often used `AvaloniaPropertyChangedEventArgs` in the Avalonia source that suggest there may be a custom property change implementation in play, but haven't looked into it.

Thank you - I was able to understand and turn your approach into a working solution within minutes!
I'll try it out in a slightly more complex scenario with a filtered tree.

If I understand correctly, the trick is to keep the messages encapsulated within the `NodeView` Component so that the host program doesn't have to re-render the `TreeView`.

Reply:
Yes, having separate component is needed here to track and update the node state and also to prevent the re-rendering of the widget tree.